src/deploy/functions/tasks.ts

```diff
@@ -192,6 +192,20 @@ export function updateFunctionTask(
       onPoll,
     };
     const cloudFunction = await pollOperation<unknown>(pollerOptions);
+    if (!backend.isEventTrigger(fn.trigger)) {
+      try {
+        if (fn.platform === "gcfv1") {
+          await gcf.setIamPolicy({
```

Comment:
This should noop if `fn.invoker` isn't set.

Reply:
Added check